src/index.ts

```
@@ -0,0 +1,170 @@
+		if (this.queueStatus === __PROCESSING__) return;
+		
+		const connectivity = await this.verifyConnectivity();
+		if (!this.queue.head || !connectivity) return;
+		
+		this.queueStatus = __PROCESSING__;
+		
+		try {
+			await this.process(this.queue.head);
+			this.dequeue();
+		} catch (err) {
+			throw err;
```

Biome

[error] The catch clause that only rethrows the original error is redundant.

These unnecessary catch clauses can be confusing. It is recommended to remove them.

(lint/complexity/noUselessCatch)